internal/buildengine/updates.go

```diff
@@ -78,7 +78,12 @@ func (u *updatesService) Ping(context.Context, *connect.Request[ftlv1.PingReques
 }
 
 func (u *updatesService) StreamEngineEvents(ctx context.Context, req *connect.Request[buildenginepb.StreamEngineEventsRequest], stream *connect.ServerStream[buildenginepb.StreamEngineEventsResponse]) error {
-	// Only send cached events if replay_history is true
+	// First subscribe to new events to avoid missing any
+	events := make(chan *buildenginepb.EngineEvent, 64)
+	u.engine.EngineUpdates.Subscribe(events)
+	defer u.engine.EngineUpdates.Unsubscribe(events)
```

Comment:
I think this is just better before we play the history. I'm not sure if this was part of the blue dots showing when the terminal is happy, but thought I'd point it out.